### ISSUE TITLE
[BUG] unhandled promise rejected on remix with vercel

### ISSUE THREAD

**Issue by pullmann4rent** [inngest/inngest-js#304]
Hello,

I use inngest with remix and vercel and have successfull deployed my app but if I send data with inngest I get this:
```
Unhandled Promise Rejection 	{"errorType":"Runtime.UnhandledPromiseRejection","errorMessage":"TypeError: First parameter has member 'readable' that is not a ReadableStream.","reason":{"errorType":"TypeError","errorMessage":"First parameter has member 'readable' that is not a ReadableStream.","stack":["TypeError: First parameter has member 'readable' that is not a ReadableStream.","    at assertReadableStream (/var/task/node_modules/web-streams-polyfill/dist/ponyfill.js:362:19)","    at convertReadableWritablePair (/var/task/node_modules/web-streams-polyfill/dist/ponyfill.js:3524:9)","    at ReadableStream.pipeThrough (/var/task/node_modules/web-streams-polyfill/dist/ponyfill.js:3608:29)","    at fetchFinale (node:internal/deps/undici/undici:10965:56)","    at mainFetch (node:internal/deps/undici/undici:10857:9)","    at processTicksAndRejections (node:internal/process/task_queues:95:5)"]},"promise":{},"stack":["Runtime.UnhandledPromiseRejection: TypeError: First parameter has member 'readable' that is not a ReadableStream.","    at process.<anonymous> (file:///var/runtime/index.mjs:1186:17)","    at process.emit (node:events:525:35)","    at process.emit (/var/task/node_modules/source-map-support/source-map-support.js:516:21)","    at emit (node:internal/process/promises:149:20)","    at processPromiseRejections (node:internal/process/promises:283:27)","    at processTicksAndRejections (node:internal/process/task_queues:96:32)"]}
Unknown application error occurred
Runtime.Unknown

```

But it send successfull the data and all works and I recieve my data after inngest is. finish but idk why I got this error.

code:
```
        if(typeof id === 'number') {

          const buffer = await getPDFVerify({db_name: connectorDB, id: id as number});
          const email = await getClientMailByID({db_name: connectorDB, p: { client_id: obj.client_id }})
        
          if(typeof email === 'string' && endObj.send_mail === '1') {
            console.log('INNGGEST');
            console.log(email);
            const saaa = await inngest.send({
              name: 'Default ingest key',
              data: {
                email,
                // @ts-ignore
                buffer
              }
            });
            console.log(saaa);
          }
        }
```
I use the latest inngest version

**Comment by jpwilliams** [comment]
Hi, @pullmann4rent! 👋

There have been a couple of reports of Remix having some issues with streams--or just generally around Node 18 compatibility and polyfills on Vercel--though this looks like it might be an issue trying to send a [`Buffer`](https://nodejs.org/docs/latest-v18.x/api/buffer.html) in the event payload.

What is the type of `buffer` (the return of `await getPDFVerify()`)? We currently only accept JSON data in event payloads.

**Comment by jpwilliams** [comment]
Going to close this as it's been quiet for a while. Feel free to re-open if you're still having this issue!

**Comment by darwin67** [comment]
@nkg168 I'm reposting some guides from previous support responses.
Not sure if it'll solve your problem completely, so could you give it a try and let us know?

### Previous answer

TL;DR I think you can work around your current problem with this:

1. Install `@remix-run/web-fetch`
2. Pass the exported `fetch` into `new Inngest()`, e.g.:
```
import { fetch as webFetch } from "@remix-run/web-fetch";
import { Inngest } from "inngest";

export const inngest = new Inngest({
  id: "my-app",
  fetch: webFetch as typeof fetch,
});
```

make sure to remove `allowStreaming: "force"` if you had that set in your `serve()` function
It looks like this is a nasty polyfilling issue that's affecting a lot of packages. Some internal checks that run (like checking if a value is a `ReadableStream`) are only partially polyfilled, resulting in some weird behaviour that's super environment dependent. It only showing up in production is a pain, eh?

- The `@vercel/remix` package has a closed issue with some continued reports affecting the convex package: [First parameter has member 'readable' that is not a ReadableStream · Issue #62 · vercel/remix (github.com)](https://github.com/vercel/remix/issues/62)
- The `@vercel/ai` package has a similar open issue: [Remix – ReadableStream type mismatch: The "transform.readable" property must be an instance of ReadableStream. Received an instance of ReadableStream · Issue #602 · vercel/ai (github.com)](https://github.com/vercel/ai/issues/602)
Interlinked from each of those are more issues, too. Facets of it seem to have been solved in Remix v2, but not everything.

For now, try the above workaround and see if it works for you. If so, I'll put a note on our docs, add an issue our side (so you can track), and add our own reproduction repo to the `@vercel/remix` issue.

**Comment by nkg168** [comment]
@darwin67 I tried the corresponding solution and it works fine 👍 
Thanks!!
![result of caller](https://github.com/inngest/inngest-js/assets/34019744/e3c573d5-6a43-44ad-89b1-9839ccc4bf9a)
![result of called](https://github.com/inngest/inngest-js/assets/34019744/3ca4ad48-a4a8-4cd9-a711-6a2ca5cb75a5)
